go/vt/vtctl/vtctl.go

```diff
@@ -2054,14 +2068,40 @@ func commandVRWorkflow(ctx context.Context, wr *wrangler.Wrangler, subFlags *fla
 	//TODO: check if invalid parameters were passed in that do not apply to this action
 	originalAction := action
 	action = strings.ToLower(action) // allow users to input action in a case-insensitive manner
+	if workflowType == wrangler.MigrateWorkflow {
+		switch action {
+		case vReplicationWorkflowActionCreate, vReplicationWorkflowActionCancel, vReplicationWorkflowActionComplete:
+		default:
+			return fmt.Errorf("invalid action for Migrate: %s", action)
+		}
+	}
+
 	switch action {
 	case vReplicationWorkflowActionCreate:
 		switch workflowType {
-		case wrangler.MoveTablesWorkflow:
+		case wrangler.MoveTablesWorkflow, wrangler.MigrateWorkflow:
+			var sourceTopo *topo.Server
+			var externalClusterName string
+
+			sourceTopo = wr.TopoServer()
 			if *sourceKeyspace == "" {
 				return fmt.Errorf("source keyspace is not specified")
 			}
-			_, err := wr.TopoServer().GetKeyspace(ctx, *sourceKeyspace)
+			if workflowType == wrangler.MigrateWorkflow {
+				splits := strings.Split(*sourceKeyspace, ".")
+				if len(splits) != 2 {
+					return fmt.Errorf("invalid format for external source cluster: %s", *sourceKeyspace)
+				}
+				externalClusterName = splits[0]
+				*sourceKeyspace = splits[1]
```

Comment:
Nit:
I feel like there should be a function `getSourceKeyspace(string) (clusterName, keyspace, err)`

Reply:
extracted that into a function